doc/TEMPLATES.md

```diff
@@ -67,6 +67,22 @@ The following template languages are supported:
 * html
 * svg
 
+### A word on Repeat, Map, and template re-rendering
+
+In web components, arrays and other complex data types are expected (if they are present at all) to be handed to the parent component as content within [slots](https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_templates_and_slots), using the contents as XML-like data (or as freestanding tags/web components), rather than on attributes.
+
+This has some subtle consequences for the built-in reactivity of web components. The main way that web components are notified of changes to their data are by updates to their attributes. This means that if you use repeat or map to render a section of your template, **changes to the arrays or objects iterated over will not automatically trigger a re-render.**
+
+As a result, you must manually trigger the `.render` method of your component somehow. The most idiomatic way to trigger a re-render would be to ensure the data you're iterating over is derived from contents within your component's slot, and then to add a listener to the [`slotchange`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSlotElement/slotchange_event) event which retriggers your component's render function:
+
+```
+static get listeners() {
+  return {slotchange: this.render}
+}
+```
+
+Another method might be to use an observation library to watch for changes to the objects you're iterating over, and then run the render function as a callback.
+
```

Comment:
@Kelketek — I think what you’re trying to highlight here is that re-rendering is _only_ triggered based on a change-by-reference detection (which is definitely by design).

I wonder if we could work this out in a separate documentation-focused PR? The implementation changes seem like they could be ready to merge pretty soon and I wouldn’t want to hold that up while we discuss the high-level documentation changes 🤙

Reply:
@theengineear Removed-- and I'll create a follow-up PR for this, probably with some revised wording. Yes, that indeed is what I'm trying to point out, but I maybe I should phrase it differently.

Reply:
@theengineear Started rewriting this, and decided I'd like to use the library a bit more before giving it another stab. I'll create the PR when I have a more complete view of what to recommend to the reader and how to frame it.